slash/frontend/slash_list.py

```diff
@@ -71,13 +72,15 @@ def slash_list(args, report_stream=sys.stdout, error_stream=sys.stderr):
             if parsed_args.only in (None, 'tests'):
                 _report_tests(parsed_args, runnables, _print)
 
+        if bool(session.warnings.warnings) and parsed_args.warnings_as_errors:
+            return 2
```

Comment:
I think it's time to either make these constants, or always return -1 on failure...